src/effects/effectchainslot.cpp

```diff
@@ -429,17 +420,75 @@ void EffectChainSlot::slotControlChainPrevPreset(double v) {
 }
 
 void EffectChainSlot::slotChannelStatusChanged(const QString& group) {
-    if (m_pEffectChain) {
-        ChannelInfo* pChannelInfo = m_channelInfoByName.value(group, NULL);
-        if (pChannelInfo != NULL && pChannelInfo->pEnabled != NULL) {
-            bool bEnable = pChannelInfo->pEnabled->toBool();
-            if (bEnable) {
-                m_pEffectChain->enableForInputChannel(pChannelInfo->handle_group);
-            } else {
-                m_pEffectChain->disableForInputChannel(pChannelInfo->handle_group);
+    ChannelInfo* pChannelInfo = m_channelInfoByName.value(group, NULL);
+    if (pChannelInfo != NULL && pChannelInfo->pEnabled != NULL) {
+        bool bEnable = pChannelInfo->pEnabled->toBool();
+        if (bEnable) {
+            enableForInputChannel(pChannelInfo->handle_group);
+        } else {
+            disableForInputChannel(pChannelInfo->handle_group);
+        }
+    }
+}
+
+void EffectChainSlot::enableForInputChannel(const ChannelHandleAndGroup& handle_group) {
+    if (m_enabledInputChannels.contains(handle_group)) {
+        return;
+    }
+
+    EffectsRequest* request = new EffectsRequest();
+    request->type = EffectsRequest::ENABLE_EFFECT_CHAIN_FOR_INPUT_CHANNEL;
+    request->pTargetChain = m_pEngineEffectChain;
+    request->EnableInputChannelForChain.pChannelHandle = &handle_group.handle();
+
+    // Allocate EffectStates here in the main thread to avoid allocating
+    // memory in the realtime audio callback thread. Pointers to the
+    // EffectStates are passed to the EffectRequest and the EffectProcessorImpls
+    // store the pointers. The containers of EffectState* pointers get deleted
+    // by ~EffectsRequest, but the EffectStates are managed by EffectProcessorImpl.
+    auto pEffectStatesMapArray = new EffectStatesMapArray;
+
+    //TODO: get actual configuration of engine
+    const mixxx::EngineParameters bufferParameters(
+          mixxx::AudioSignal::SampleRate(96000),
+          MAX_BUFFER_LEN / mixxx::kEngineChannelCount);
+
+    for (int i = 0; i < m_effects.size(); ++i) {
```

Comment:
This should be as simple as looping over all the EffectSlots and calling a method for them to create an EffectState for the input channel. I started working on this, but unfortunately it is a bit of a pain right now and will require more work later because of the separation between EffectSlot and Effect. For now let's leave it as a TODO.